### PR TITLE
Fixes Wingdings

### PR DESCRIPTION
**Pull request by kugamo**
## What Does This PR Do
This PR fixes wingdings chat messages being "" or rarely ever working.

## Why It's Good For The Game
Having features of the game work is good.

## Images of changes
Q25QQMst5V

## Changelog
:cl:
fix: Fixes Wingdings chat messages not working.
/:cl:

### CODE REVIEW

File: code/game/dna/genes/disabilities.dm [hunk 1]
```diff
@@ -249,14 +249,14 @@
 	block = GLOB.wingdingsblock
 
 /datum/dna/gene/disability/wingdings/OnSay(mob/M, message)
-	var/list/chars = string2charlist(message)
 	var/garbled_message = ""
-	for(var/C in chars)
-		if(C in GLOB.alphabet_uppercase)
+	var/messagelength = length(message)
+	for(var/i=1;i<=messagelength;i++)
```
Comment (farie82) [inline code/game/dna/genes/disabilities.dm:254]: spacing here

**Review by farie82 — CHANGES_REQUESTED** [review]
Can also remove `string2charlist` now since it's not used anymore and rather broken

File: code/game/dna/genes/disabilities.dm [hunk 2]
```diff
@@ -249,14 +249,14 @@
 	block = GLOB.wingdingsblock
 
 /datum/dna/gene/disability/wingdings/OnSay(mob/M, message)
-	var/list/chars = string2charlist(message)
 	var/garbled_message = ""
-	for(var/C in chars)
-		if(C in GLOB.alphabet_uppercase)
+	var/messagelength = length(message)
+	for(var/i = 1; i <= messagelength; i++)
```
Comment (AffectedArc07) [inline code/game/dna/genes/disabilities.dm:254]: This style of loop is slower
```dm
/proc/test1()
    for(var/i = 0; i < 500000; i++)
        var/e = i

/proc/test2()
    for(var/i in 1 to 500000)
        var/e = i

/proc/main()
    BENCHT("1", 1, test1())
    BENCHT("2", 1, test2())
```
Outputs
```
1: 57.857 ms
2: 26.734 ms
```

Please use something along the lines of
`for(var/i in 1 to length(message))`

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/game/dna/genes/disabilities.dm (hunk 2 above)
Comment (kugamo) [reply code/game/dna/genes/disabilities.dm]: done

**Review by farie82 — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**